v2/rust/src/config.rs

```diff
@@ -0,0 +1,80 @@
+use serde::Deserialize;
+use serde_json::from_str;
+use std::collections::HashMap;
+use std::fs::read_to_string;
+use std::path::{Path, PathBuf};
+
+pub fn load(path: &Path) -> anyhow::Result<Config> {
+    Ok(from_str(&read_to_string(path)?)?)
+}
+
+#[derive(Deserialize)]
+pub struct Config {
+    pub working_directory: PathBuf,
+    pub results_directory: PathBuf,
+    suites: HashMap<String, SuiteConfig>,
+}
+
+#[derive(Deserialize)]
+pub struct SuiteConfig {
+    pub robot_framework_config: RobotFrameworkConfig,
+    pub execution_config: ExecutionConfig,
+    pub environment_config: EnvironmentConfig,
+    pub session_config: SessionConfig,
+}
+
+#[derive(Deserialize)]
+pub struct RobotFrameworkConfig {
+    pub robot_target: PathBuf,
+    pub variable_file: Option<PathBuf>,
+    pub argument_file: Option<PathBuf>,
+    pub retry_strategy: RetryStrategy,
+}
+
+#[derive(Deserialize)]
+pub enum RetryStrategy {
+    Incremental,
+    Complete,
+}
+
+#[derive(Deserialize)]
+pub struct ExecutionConfig {
+    pub n_retries_max: usize,
+    pub execution_interval_seconds: u64,
+    pub timeout: u64,
+}
+
+#[derive(Deserialize)]
+#[serde(tag = "type")]
+pub enum EnvironmentConfig {
+    System,
+    Rcc(RCCEnvironmentConfig),
+}
+
+#[derive(Deserialize)]
+pub struct RCCEnvironmentConfig {
+    pub binary_path: PathBuf,
+    pub robocorp_home_path: PathBuf,
+    pub robot_yaml_path: PathBuf,
+    pub build_timeout: u64,
+}
+
+#[derive(Deserialize)]
+#[serde(tag = "type")]
+pub enum SessionConfig {
+    Current,
+    SpecificUser(UserSessionConfig),
+}
+
+#[derive(Deserialize)]
+pub struct UserSessionConfig {
+    pub user_name: String,
+}
+
+impl Config {
+    pub fn suites(&self) -> Vec<(&String, &SuiteConfig)> {
```

Comment:
It's very difficult to review this, since it is not used anywhere.

Reply:
It will be used in follow-up commits. The purpose is to make sure that the scheduler always builds environments in the same order. Would you prefer if I added this later on?

Reply:
OTI: Interesting, but I guess the building of the environments is part of the test run? I guess this ensures the tests are started in the same order? The building the hololib library can be ensured to be somewhat deterministic in this manner I suppose.
If this is business logic, then it maybe warrants a test.

Reply:
I added a test